python/lsst/ctrl/bps/wms/htcondor/htcondor_service.py

```diff
@@ -309,3 +305,520 @@ def handle_job_inputs(generic_workflow: GenericWorkflow, job_name: str,
+                   jobs[wms_workflow_id]["total_jobs"])
+    except StopIteration:
+        message = f"Could not find HTCondor files in {wms_path}"
+        _LOG.warning(message)
+        wms_workflow_id = MISSING_ID
+        jobs = {}
+
+    return wms_workflow_id, jobs, message
+
+
+def _create_detailed_report_from_jobs(wms_workflow_id, jobs):
+    """Gather run information to be used in generating summary reports.
+
+    Parameters
+    ----------
+    wms_workflow_id : `str`
+        Run lookup restricted to given user.
+    jobs : `float`
+        How many previous days to search for run information.
+
+    Returns
+    -------
+    run_reports : `dict` [`str`, `WmsRunReport`]
+        Run information for the detailed report.  The key is the given HTCondor id
+        and the value is a collection of report information for that run.
+    message : `str`
+        Message to be printed with the summary report.
+    """
+    _LOG.debug("_create_detailed_report: id = %s, job = %s", wms_workflow_id, jobs[wms_workflow_id])
+    dag_job = jobs[wms_workflow_id]
+    if "total_jobs" not in dag_job or "DAGNodeName" in dag_job:
+        _LOG.error("Job ID %s is not a DAG job.", wms_workflow_id)
+        return {}
+    report = WmsRunReport(wms_id=wms_workflow_id,
+                          path=dag_job["Iwd"],
+                          label=dag_job.get("bps_job_label", "MISS"),
+                          run=dag_job.get("bps_run", "MISS"),
+                          project=dag_job.get("bps_project", "MISS"),
+                          campaign=dag_job.get("bps_campaign", "MISS"),
+                          payload=dag_job.get("bps_payload", "MISS"),
+                          operator=_get_owner(dag_job),
+                          run_summary=_get_run_summary(dag_job),
+                          state=_htc_status_to_wms_state(dag_job),
+                          jobs=[],
+                          total_number_jobs=dag_job['total_jobs'],
+                          job_state_counts=dag_job['state_counts'])
+
+    try:
+        for job in jobs.values():
+            if job["ClusterId"] != int(float(wms_workflow_id)):
+                job_report = WmsJobReport(wms_id=job["ClusterId"],
+                                          name=job["DAGNodeName"],
+                                          label=job.get("bps_job_label",
+                                                        pegasus_name_to_label(job["DAGNodeName"])),
+                                          state=_htc_status_to_wms_state(job))
+                if job_report.label == "init":
+                    job_report.label = "pipetaskInit"
+                report.jobs.append(job_report)
+    except KeyError as ex:
+        _LOG.error("Job missing key '%s': %s", str(ex), job)
+        raise
+
+    run_reports = {report.wms_id: report}
+    _LOG.debug("_create_detailed_report: run_reports = %s", run_reports)
+    return run_reports
+
+
+def summary_report(user, hist, pass_thru):
```

Comment:
Why doesn't it start with the leading underscore like the others? It's not included in `__all__` and it doesn't look like it's used outside of this module anyway.